notary-plugins/notary-plugin-common/src/main/kotlin/com/r3/corda/notary/plugin/common/PluggableNotaryFlowHelpers.kt

```diff
@@ -0,0 +1,117 @@
+fun PublicKey.toBase58String(): String = this.encoded.toBase58()
+
+/** Verifies that the correct notarisation request was signed by the counterparty. */
+@Suspendable
+fun validateRequestSignature(notarisationRequest: NotarisationRequest,
+                             requestingParty: Party,
+                             serializationService: SerializationService,
+                             signatureVerifier: DigitalSignatureVerificationService,
+                             signature: NotarisationRequestSignature
+) {
+    val digitalSignature = signature.digitalSignature
+    require(requestingParty.owningKey == digitalSignature.by) {
+        "Expected a signature by ${requestingParty.owningKey.toBase58String()}, " +
+                "but received by ${digitalSignature.by.toBase58String()}}"
+    }
+
+    // TODO CORE-3698: Review if this TODO is still valid
+    //  if requestSignature was generated over an old version of NotarisationRequest, we need to be able to
+    //  reserialize it in that version to get the exact same bytes. Modify the serialization logic once that's
+    //  available.
+    val expectedSignedBytes = serializationService.serialize(notarisationRequest).bytes
+    signatureVerifier.verify(
+        digitalSignature.by,
+        SignatureSpec.ECDSA_SHA256, // TODO This shouldn't be hardcoded?
+        digitalSignature.bytes,
+        expectedSignedBytes
+    )
+}
+
+/** Creates a signature over the notarisation request using the legal identity key. */
+@Suspendable
+fun generateRequestSignature(notarisationRequest: NotarisationRequest,
+                             memberInfo: MemberInfo,
+                             serializationService: SerializationService,
+                             signingService: SigningService
+): NotarisationRequestSignature {
+    val serializedRequest = serializationService.serialize(notarisationRequest).bytes
+    val myLegalIdentity = memberInfo.sessionInitiationKey
+    val signature = signingService.sign(
+        serializedRequest,
+        myLegalIdentity,
+        SignatureSpec.ECDSA_SHA256 // TODO This shouldn't be hardcoded?
+    )
+    return NotarisationRequestSignature(signature, memberInfo.platformVersion)
+}
+
+/**
+ * A helper function that will convert a [UniquenessCheckResponse] to a [NotarisationResponse].
+ */
+@Suspendable
+fun UniquenessCheckResponse.toNotarisationResponse(): NotarisationResponse {
+    val (signature, notaryError) = when (val uniquenessResult = result) {
+        is UniquenessCheckResultSuccess -> Pair(
+            listOf(signature!!),
+            null
+        )
+        is UniquenessCheckResultFailure -> Pair(
+            emptyList(),
+            uniquenessResult.error.toNotaryError()
+        )
+        else -> Pair(
+            emptyList(),
+            NotaryErrorGeneralImpl(
+                "Unknown uniqueness check result: $uniquenessResult"
+            )
+        )
+    }
+    return NotarisationResponse(signature, notaryError)
```

Comment:
```suggestion
    return when (val uniquenessResult = result) {
        is UniquenessCheckResultSuccess -> NotarisationResponseImpl(
            listOf(signature!!),
            null
        )
        is UniquenessCheckResultFailure -> NotarisationResponseImpl(
            emptyList(),
            uniquenessResult.error.toNotaryError()
        )
        else -> NotarisationResponseImpl(
            emptyList(),
            NotaryErrorGeneralImpl(
                "Unknown uniqueness check result: $uniquenessResult"
            )
        )
    }
```

Bit more succinct, and follows the pattern of the similar `toNotaryError` function below